### PR TITLE
Add FileSystem::remove

### PR DESCRIPTION
**Pull request by oerling**
Add a signature for deleting a file and an implementation for local
file system.

### CODE REVIEW

**Review by mbasmanova — COMMENTED** [review]
@oerling Looks good, but PR title and description need updating. They contain some strange text.

File: velox/common/file/FileSystems.h [hunk 1]
```diff
@@ -41,6 +43,11 @@ class FileSystem {
   virtual std::unique_ptr<WriteFile> openFileForWrite(
       std::string_view path) = 0;
 
+  // Deletes the file at 'path'. Throws on error.
+  virtual void remove(std::string_view path) {
```
Comment (mbasmanova) [inline velox/common/file/FileSystems.h:47]: Would it be cleaner to make this method pure virtual and have each implementation decide whether to support it or not?

**Comment by majetideepak** [comment]
Changes look good to me as well. Same thought if "remove" should be a pure virtual function as Masha mentioned.

**Review by majetideepak — APPROVED** [review]
+1

**Review by frankobe — APPROVED** [review]

**Pull request closed without merging**